game/src/contrib/entities/AIFactory.java

```diff
@@ -0,0 +1,215 @@
+ * NOTE: If the first Monster rolls a ProtectOn AI, it will Protect itself.
+ */
+public class AIFactory {
+    //FightAI Parameters:
+    //CollideAI
+    private static final float rushRangeLow = 0.5f;
+    private static final float rushRangeHigh = 2.0f;
+
+    //RangeAI
+    private static final float attackRangeLow = 2.1f;
+    private static final float attackRangeHigh = 8f;
+    private static final float distanceLow = 1f;
+    private static final float distanceHigh = 2f;
+
+
+    //IdleAi Parameters:
+    //PatroullieWalk
+    private static final float patrouilleRadiusLow = 2f;
+    private static final float patrouilleRadiusHigh = 6f;
+    private static final int checkpointsLow = 2;
+    private static final int checkpointsHigh = 6;
+    private static final int pauseTimeLow = 1;
+    private static final int pauseTimeHigh = 5;
+
+    //RadiusWalk
+    private static final float radiusWalkLow = 2f;
+    private static final float radiusWalkHigh = 8f;
+    private static final int breakTimeLow = 1;
+    private static final int breakTimeHigh = 5;
+
+    //StaticRadiusWalk
+    private static final float staticRadiusWalkLow = 2f;
+    private static final float staticRadiusWalkHigh = 8f;
+    private static final int staticBreakTimeLow = 1;
+    private static final int staticBreakTimeHigh = 5;
+
+
+    //TransitionAI Parameters:
+    //RangeTransition
+    private static final float rangeTransitionLow = 2f;
+    private static final float rangeTransitionHigh = 10f;
+
+    //ProtectOnApproach
+    private static final float protectRangeLow = 2f;
+    private static final float protectRangeHigh = 8f;
+
+
+
+
+
+    /**
+     * Constructs a random FightAI with random parameters.
+     * Needs to be Updated whenever a new FightAI is added.
+     * @return the generated FightAI
+     * NOTE: MeleeAI currently gets excluded from generation due to a bug that crashes the Game.
+     * The MeleeAI seems to have trouble with Pathing as it often has null as its path, causing the crash.
+     */
+    public static Consumer<Entity> generateRandomFightAI(){
+        Random random = new Random();
+        int index = random.nextInt(0,2);
+
+        return switch (index) {
+            case 0 -> new CollideAI(
+                random.nextFloat(rushRangeLow, rushRangeHigh)
+            );
+            case 1 -> new RangeAI(
+                random.nextFloat(attackRangeLow, attackRangeHigh),
+                random.nextFloat(distanceLow,distanceHigh),
+                new Skill(
+                    new FireballSkill(SkillTools::getHeroPositionAsPoint),
+                    1)
+            );
+            /*
+            case 2 -> new MeleeAI(
+                1f,
+                new Skill(
+                    new FireballSkill(SkillTools::getHeroPositionAsPoint),
+                    1)
+            );*/
+            default -> throw new IndexOutOfBoundsException("This IdleAI does not exist!");
```

Comment:
Ist das nicht "non reachable code"? 
In Zeile 90 erzeugt ihr eine Zahl die entweder 0 oder 1 ist und beide Szenarien deckt ihr ab.

Reply:
An sich schon, leider bekomme ich da einen Error da ich nicht "alle fälle abdecke". Hab ich erstmal drinne gelassen.